src/custom_nodes/east_ocr/east_ocr.cpp

```diff
@@ -0,0 +1,399 @@
+            cv::Point2i p3{-cos * w + p2.x, sin * w + p2.y};
+            cv::Point2i p4{p3.x + p1.x - p2.x, p3.y + p1.y - p2.y};
+
+            int x1 = std::min(std::min(std::min(p2.x, p1.x), p3.x), p4.x);
+            int x2 = std::max(std::max(std::max(p2.x, p1.x), p3.x), p4.x);
+            int y1 = std::min(std::min(std::min(p2.y, p1.y), p3.y), p4.y);
+            int y2 = std::max(std::max(std::max(p2.y, p1.y), p3.y), p4.y);
+
+            if (debugMode) {
+                std::cout << "Angled polygon coordinates: " << std::endl;
+                std::cout << p4 << p3 << p1 << p2 << std::endl;
+
+                std::cout << "Polygon bounding box with no rotation: " << std::endl;
+                std::cout << cv::Point2i(x1, y1) << cv::Point2i(x2, y2) << std::endl;
+                std::cout << "---------------------------" << std::endl;
+            }
+
+            rects.emplace_back(x1, y1, x2 - x1 - 1, y2 - y1 - 1);
+            scores.emplace_back(score);
+        }
+    }
+
+    if (debugMode)
+        std::cout << "Total findings: " << rects.size() << std::endl;
+
+    std::vector<cv::Rect> filteredBoxes;
+    std::vector<float> filteredScores;
+    nms2(rects, scores, filteredBoxes, filteredScores, overlapThreshold);
+    NODE_ASSERT(filteredBoxes.size() == filteredScores.size(), "filtered boxes and scores are not equal length");
+    if (filteredBoxes.size() > maxOutputBatch) {
+        filteredBoxes.resize(maxOutputBatch);
+        filteredScores.resize(maxOutputBatch);
+    }
+
+    if (debugMode)
+        std::cout << "Total findings after NMS2 (non max suppression) filter: " << filteredBoxes.size() << std::endl;
+
+    *outputsCount = 3;
+    *outputs = (struct CustomNodeTensor*)malloc(*outputsCount * sizeof(CustomNodeTensor));
+
+    CustomNodeTensor& textImagesTensor = (*outputs)[0];
+    textImagesTensor.name = TEXT_IMAGES_TENSOR_NAME;
+    if (extract_text_images_into_output(&textImagesTensor, filteredBoxes, image, targetImageHeight, targetImageWidth, convertToGrayScale, imageLayout)) {
+        return 1;
+    }
+
+    CustomNodeTensor& coordinatesTensor = (*outputs)[1];
+    coordinatesTensor.name = COORDINATES_TENSOR_NAME;
+    if (extract_coordinates_into_output(&coordinatesTensor, filteredBoxes)) {
+        return 1;
+    }
+
+    CustomNodeTensor& confidenceTensor = (*outputs)[2];
+    confidenceTensor.name = CONFIDENCE_TENSOR_NAME;
+    if (extract_scores_into_output(&confidenceTensor, filteredScores)) {
+        return 1;
+    }
+
+    return 0;
+}
+
+int getInputsInfo(struct CustomNodeTensorInfo** info, int* infoCount, const struct CustomNodeParam* params, int paramsCount) {
+    int originalImageHeight = get_int_parameter("original_image_height", params, paramsCount, -1);
+    int originalImageWidth = get_int_parameter("original_image_width", params, paramsCount, -1);
+    NODE_ASSERT(originalImageHeight > 0, "original image height must be larger than 0");
```

Comment:
I think part of the NODE_ASSERTS can be shared across 3 entry points to custom node and then just just sth like
retcode = checkParams(params);

Reply:
This way we have multiple places in code for parsing parameters which would confuse the reader